pkg/util/lifted/scheduler/cache/node_tree_test.go

```diff
@@ -0,0 +1,421 @@
+			newNode:       makeNodeWithTopology("node1", "us-east", "us-east-1b"),
+			expectedZones: []string{getExpectedZoneKey("us-east", "us-east-1b")},
+			expectedCount: 1,
+		},
+		{
+			name: "update from unlabeled to labeled",
+			existingNodes: []*corev1.Node{
+				{ObjectMeta: metav1.ObjectMeta{Name: "node1"}},
+			},
+			oldNode:       &corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: "node1"}},
+			newNode:       makeNodeWithTopology("node1", "us-east", "us-east-1a"),
+			expectedZones: []string{getExpectedZoneKey("us-east", "us-east-1a")},
+			expectedCount: 1,
+		},
+		{
+			name: "update from labeled to unlabeled",
+			existingNodes: []*corev1.Node{
+				makeNodeWithTopology("node1", "us-east", "us-east-1a"),
+			},
+			oldNode:       makeNodeWithTopology("node1", "us-east", "us-east-1a"),
+			newNode:       &corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: "node1"}},
+			expectedZones: []string{""},
+			expectedCount: 1,
+		},
+		{
+			name:          "update non-existent node",
+			existingNodes: []*corev1.Node{},
+			oldNode:       makeNodeWithTopology("node1", "us-east", "us-east-1a"),
+			newNode:       makeNodeWithTopology("node1", "us-east", "us-east-1b"),
+			expectedZones: []string{getExpectedZoneKey("us-east", "us-east-1b")},
+			expectedCount: 1,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nt := newNodeTree(tt.existingNodes)
+			nt.updateNode(tt.oldNode, tt.newNode)
+
+			assert.ElementsMatch(t, tt.expectedZones, nt.zones)
+			assert.Equal(t, tt.expectedCount, nt.numNodes)
+		})
+	}
+}
+
+func TestList(t *testing.T) {
+	tests := []struct {
+		name        string
+		nodes       []*corev1.Node
+		expectedLen int
+	}{
+		{
+			name:        "empty tree",
+			nodes:       []*corev1.Node{},
+			expectedLen: 0,
+		},
+		{
+			name: "single zone",
+			nodes: []*corev1.Node{
+				makeNodeWithTopology("node1", "us-east", "us-east-1a"),
+				makeNodeWithTopology("node2", "us-east", "us-east-1a"),
+			},
+			expectedLen: 2,
+		},
+		{
+			name: "multiple zones",
+			nodes: []*corev1.Node{
+				makeNodeWithTopology("node1", "us-east", "us-east-1a"),
+				makeNodeWithTopology("node2", "us-east", "us-east-1b"),
+				makeNodeWithTopology("node3", "us-east", "us-east-1a"),
+			},
+			expectedLen: 3,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nt := newNodeTree(tt.nodes)
+			list, err := nt.list()
+
+			assert.NoError(t, err)
+			assert.Equal(t, tt.expectedLen, len(list))
+
+			// Verify that all nodes are in the list exactly once
+			nodeCount := make(map[string]int)
+			for _, nodeName := range list {
+				nodeCount[nodeName]++
+			}
+
+			// Verify each node appears exactly once and all nodes are present
+			for _, node := range tt.nodes {
+				count, exists := nodeCount[node.Name]
+				assert.True(t, exists, "node %s should be in the list", node.Name)
+				assert.Equal(t, 1, count, "node %s should appear exactly once", node.Name)
```

Comment:
The `nodeTree` has no restriction that `nodeName` cannot be duplicated, so determining whether `count` is 1 may be a bit inappropriate.

How about directly determine if the result of `nt.list()` is the expected slice? Like 

```go
func TestList(t *testing.T) {
	tests := []struct {
		name         string
		nodes        []*corev1.Node
		expectedList []string
	}{
		{
			name: "multiple zones",
			nodes: []*corev1.Node{
				makeNodeWithTopology("node1", "us-east", "us-east-1a"),
				makeNodeWithTopology("node2", "us-east", "us-east-1b"),
				makeNodeWithTopology("node3", "us-east", "us-east-1a"),
				makeNodeWithTopology("node4", "us-east", "us-east-1c"),
			},
			expectedList: []string{"node1", "node2", "node3", "node4"},
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			nt := newNodeTree(tt.nodes)
			list, err := nt.list()
			assert.NoError(t, err)
			sort.Strings(tt.expectedList)
			sort.Strings(list)
			assert.Equal(t, tt.expectedList, list)
		})
	}
}
```